index.js

```diff
@@ -160,7 +136,7 @@ Ycb.prototype = {
                     console.log(JSON.stringify(this.settings[lookupPaths[path]], null, 4));
                 }
                 // merge a copy so that we don't modify the source
-                config = objectMerge(this._cloneObj(this.settings[lookupPaths[path]]), config);
+                config = mergeDeep(this.settings[lookupPaths[path]], config);
```

Comment:
This is the line that gets optimized: instead of sweeping across the object twice (once for clone, once for merge), it only happens once and `mergeDeep` ensures that objects are not referenced from the resulting object.